### PR TITLE
[WIP] StringSource#originalPositionFromPosition returns an incorrect position

### PR DESCRIPTION
**Pull request by nodaguti**
While `column` is the 1-based index according to [the textlint's documentation](https://github.com/textlint/textlint#use-as-node-modules), I've noticed that some tests regarding `originalPositionFromPosition` specified `index` (0-based index) as `column` (1-based index).

This PR fixes such tests, but currently one test fails with these changes, so I guess `originalPositionFromPosition` returns a wrong column value in some edge cases.

```
  1) StringSource #originalPositionFromPosition Str + Link:

        # test/StringSource-test.js:238

  assert.deepEqual(source.originalPositionFromPosition({ line: 1, column: 8 }), { line: 1, column: 8 })
                   |      |                            |                        |
                   |      Position{line:1,column:9}    Object{line:1,column:8}  Object{line:1,column:8}
                   StringSource{rootNode:#Object#,tokenMaps:#Array#,generatedString:"This is Example！？",originalSource:#StructuredSource#,generatedSource:#StructuredSource#}

      + expected - actual

       {
      -  "column": 9
      +  "column": 8
         "line": 1
       }

  AssertionError: Position { line: 1, column: 9 } deepEqual { line: 1, column: 8 }
      at Decorator._callFunc (node_modules/empower-core/lib/decorator.js:110:20)
      at Decorator.concreteAssert (node_modules/empower-core/lib/decorator.js:103:17)
      at Function.decoratedAssert [as deepEqual] (node_modules/empower-core/lib/decorate.js:49:30)
      at Context.<anonymous> (StringSource-test.js:238:20)
```

I'm sorry I'm yet to narrow the cause of this mismatch.

Do you have any thoughts or suggestions?

### CODE REVIEW

File: test/StringSource-test.js [hunk 1]
```diff
@@ -234,13 +234,16 @@ describe("StringSource", function() {
...
             let result = source.toString();
             assert.equal(result, "This is Example！？");
+            // The second whitespace
             assert.deepEqual(source.originalPositionFromPosition({
                 line: 1,
                 column: 8
             }), {
                 line: 1,
-                column: 9
+                column: 8
```
Comment (nodaguti) [inline test/StringSource-test.js]: Because the position `(1, 8)` in the generated sentence represents the second whitespace, which is before a link expression, its position in the original sentence should also be `(1, 8)`.

File: test/StringSource-test.js [hunk 2]
```diff
@@ -255,15 +258,16 @@ describe("StringSource", function() {
             let source = new StringSource(AST);
             let result = source.toString();
             assert.equal(result, "First\nalt text");
-            // 4
```
Comment (nodaguti) [inline test/StringSource-test.js]: Removed this comment as I'm not sure what this comment means.

File: test/StringSource-test.js [hunk 3]
```diff
@@ -255,15 +258,16 @@ describe("StringSource", function() {
...
             var lines = result.split("\n");
             var indexOf = lines[1].indexOf("text");
+            // text
+            // ^
             assert.deepEqual(source.originalPositionFromPosition({
-                line: lines.length,
-                column: indexOf
+                line: 2,
```
Comment (nodaguti) [inline test/StringSource-test.js]: Changed to the numeric literal to be consistent with other test cases.

File: test/StringSource-test.js [hunk 4]
```diff
@@ -255,15 +258,16 @@ describe("StringSource", function() {
...
+                column: indexOf + 1
```
Comment (nodaguti) [inline test/StringSource-test.js]: `String#indexOf` returns 0-based index though `column` here expects 1-based index.

**Comment by nodaguti** [comment]
I'm sorry for confusing you, but I've found that `column` in `textlint-util-to-string` is actually 0-based, not like `textlint` itself.
